docs/specs/openapi.yaml

```
@@ -136,18 +341,227 @@ paths:
           content:
             application/json:
               schema:
-                $ref: "#/components/schemas/GetChecks"
-              examples:
-                get checks:
-                  value:
-                    type: sync
-                    status-code: 200
-                    status: OK
-                    result:
-                      - name: svc1-up
-                        status: up
-                        threshold: 3
-                        change-id: 91
+                $ref: "#/components/schemas/GetChecksResponse"
+              example:
+                {
+                  "type": "sync",
+                  "status-code": 200,
+                  "status": "OK",
+                  "result": [
+                    {
+                      "name": "check1",
+                      "status": "up",
+                      "threshold": 3,
+                      "change-id": "37"
+                    }
+                  ]
+                }
+  /v1/files:
+    get:
+      summary: Read or list files
+      tags:
+        - files
+      description: Read the contents of files or list files from the remote system.
+      parameters:
+        - name: action
+          in: query
+          description: Action to perform.
+          required: true
+          schema:
+            type: string
+            enum: ["list", "read"]
+        - name: path
+          in: query
+          description: |
+            For "read": Absolute file path to read. To read multiple files, specify this parameter multiple times.
+            
+            For "list": Absolute path to the directory to list.
+          required: true
+          schema:
+            type: string
+          style: form # For handling comma-separated values in "read".
+          explode: true # For handling comma-separated values in "read".
+        - name: pattern
+          in: query
+          description: Glob pattern to filter files/directories for the "list" action.
+          schema:
+            type: string
+        - name: itself
+          in: query
+          description: Whether to list the directory itself, rather than its contents for the "list" action.
```

At first I didn't understand what "list the directory itself" meant. How about this wording:
```suggestion
          description: For the "list" action, `itself` specifies whether to return information about the directory itself ("true") or list the contents of the directory ("false").
```

---

Yes, here I should not have invented this description myself, it is not clear enough.

While I found a [comment from the code](https://github.com/canonical/pebble/blob/master/client/files.go#L44) on this parameter, it is not as good as the version you proposed in my eyes. So here I will use what you proposed.